zabbix/mysql/schema.sls

```diff
@@ -5,25 +5,62 @@
 {% set dbname = settings.get('dbname', 'zabbix') -%}
 {% set dbuser = settings.get('dbuser', 'zabbixuser') -%}
 {% set dbpass = settings.get('dbpass', 'zabbixpass') -%}
+{% set dbroot_user = settings.get('dbroot_user') -%}
+{% set dbroot_pass = settings.get('dbroot_pass') -%}
+
+{% if zabbix.version_repo|float < 3 -%}
+{% set zabbix_db_ver = '22' -%}
+{% else %}
+{% set zabbix_db_ver = '34' -%}
+{% endif -%}
 
 
 include:
   - zabbix.mysql.conf
 
 
-{% for file in [
-  '/usr/share/zabbix-server-mysql/salt-provided-schema.sql',
-  '/usr/share/zabbix-server-mysql/salt-provided-images.sql',
-  '/usr/share/zabbix-server-mysql/salt-provided-data.sql'
-] -%}
-{{ file }}:
-  file.managed:
-    - makedirs: True
-    - source: {{ files_switch('zabbix', [ file ]) }}
-  cmd.run:
-    - name: /usr/bin/mysql -h {{ dbhost }} -u {{ dbuser }} --password={{ dbpass }} {{ dbname }} < {{ file }} && touch {{ file }}.applied
-    - unless: test -f {{ file }}.applied
-    - require:
-      - file: {{ file }}
-      - pkg: mysql-client
-{% endfor -%}
+/usr/share/zabbix-server-mysql:
+  file.recurse:
+    - source: salt://zabbix/files/default/usr/share/zabbix-server-mysql/{{ zabbix_db_ver }}
+    - onchanges:
```

Comment:
I was building off of #55.  The onchanges requisite was used to prevent multiple executions of the SQL files after the initial run (as opposed to using the touch commands to add `.applied`).  I'll take a look at your PR and see what might make the most sense.  I'm not sure yet how best to handle changing between versions (e.g., upgrading from 3.0 to 3.4).

Reply:
Zabbix server can upgrade DB schema by itself. So if DB is present and contain some tables we can assume that upload of new data is not required. I think [salt.modules.mysql.db_tables](https://docs.saltstack.com/en/latest/ref/modules/all/salt.modules.mysql.html#salt.modules.mysql.db_tables) can be used for that check.